### DISCUSSION TITLE
onComplete method need to wait for API call response

### DISCUSSION THREAD

**Discussion by DapperNerd** [vydimitrov/react-countdown-circle-timer#161] [Q&A] [answered]
Hello team,

I trying to use the countdown timer in one of my project. But facing one issue. There is a method called onComplete with return an array [true/false(to continue timer or stop it), 1500(add some delay)].

The issue faced by me is I am making an API call as timer completed and before returning the array I want to wait for api response. For this I am using async await in react. Below is code for reference 

![image](https://user-images.githubusercontent.com/28612942/128025285-7a8e4618-ec09-49d1-8854-bf67dd59889f.png)

But I am getting the following error need help if you can guide me here
![image](https://user-images.githubusercontent.com/28612942/128025444-da3aa177-674d-41c1-92fa-46c86a1317fe.png)

**Comment by vydimitrov** [comment, marked as answer]
Hey, yup `async await`  returns `Promise`, which breaks the code since `onComplete` expects either an array or nothing as a returned value. You can do the following:

```js
const [key, setKey] = useState(0)

.
.
.

<CountdownTimer
  key={key}
  onCopmplete={() => {
     (async () => {
        // do your async stuff here
       // if you need to repeat the timer just set the key and the timer will start over again
      setKey(prev => prev + 1)
     })()
  }}
/>
```
`(async() => {})()` is [this one](https://usefulangle.com/post/248/javascript-async-anonymous-function-iife#:~:text=Immediately%2Dinvoked%20%2F%20anonymous%20async%20functions%20are%20a%20way%20to%20use,without%20polluting%20the%20global%20namespace.)

**Reply by DapperNerd** [reply]
Hello @vydimitrov Thank you for the response this is exactly what I was looking for and it solves my problem I was facing.

**Reply by vydimitrov** [reply]
Cool, I am happy it works for you.
